streamer/src/nonblocking/connection_rate_limiter.rs

```diff
@@ -0,0 +1,53 @@
+use {
+    governor::{DefaultDirectRateLimiter, DefaultKeyedRateLimiter, Quota, RateLimiter},
+    std::{net::IpAddr, num::NonZeroU32},
+};
+
+pub struct ConnectionRateLimiter {
+    limiter: DefaultKeyedRateLimiter<IpAddr>,
+}
+
+impl ConnectionRateLimiter {
+    pub fn new(limit_per_minute: u32) -> Self {
+        let quota = Quota::per_minute(NonZeroU32::new(limit_per_minute).unwrap()); // Adjust the rate limit as needed
+        Self {
+            limiter: DefaultKeyedRateLimiter::keyed(quota),
+        }
+    }
+
+    pub fn check(&self, ip: &IpAddr) -> bool {
```

Comment:
can this have a better self-documenting name, `should_limit()` or something?

Reply:
i'm not sure this comment is helpful. "as needed"???

Reply:
Changed comments.